src/foam/util/SafetyUtil.java

```diff
@@ -12,9 +12,13 @@ public static int compare(Object o1 , Object o2) {
     if ( o1 == null ) return -1;
 
     if (o1 instanceof Number && o2 instanceof Number) {      
-      if( ((Number) o1).doubleValue() == ((Number) o2).doubleValue()) return 0;
-      if( ((Number) o1).doubleValue() >  ((Number) o2).doubleValue()) return 1;
-      if( ((Number) o1).doubleValue() <  ((Number) o2).doubleValue()) return -1;
+      Double d1 =((Number) o1).doubleValue();
```

Comment:
double not Double, Double is a slow fat Object whereas double is a small fast native type